src/components/Company/DocumentSigner/SignatureForm/SignatureForm.tsx

```diff
@@ -0,0 +1,122 @@
+import { ReactNode } from 'react'
+import { useI18n } from '@/i18n'
+import {
+  BaseComponent,
+  BaseComponentInterface,
+  createCompoundContext,
+  useBase,
+} from '@/components/Base/Base'
+import {
+  SignatureForm as SharedSignatureForm,
+  type SignatureFormInputs,
+} from '@/components/Common/SignatureForm'
+import { useSignCompanyForm, useGetCompanyFormPdf } from '@/api/queries/companyForms'
+import { Flex } from '@/components/Common'
+import { type Schemas } from '@/types/schema'
+import { companyEvents } from '@/shared/constants'
+
+import { Head } from './Head'
+import { Preview } from './Preview'
+import { Form } from './Form'
+import { Actions } from './Actions'
+
+type SignatureFormContextType = {
+  form: Schemas['Form']
+  pdfUrl?: string
+  isPending: boolean
+  onBack: () => void
+}
+
+const [useSignatureForm, SignatureFormProvider] = createCompoundContext<SignatureFormContextType>(
+  'CompanySignatureFormContext',
+)
+
+export { useSignatureForm }
+
+interface SignatureFormProps {
+  form: Schemas['Form']
+  companyId: string
+  children?: ReactNode
+  className?: string
+}
+
+export function SignatureForm({
+  form,
+  companyId,
+  children,
+  className,
+  ...props
+}: SignatureFormProps & BaseComponentInterface) {
+  return (
+    <BaseComponent {...props}>
+      <Root form={form} companyId={companyId} className={className}>
+        {children}
+      </Root>
+    </BaseComponent>
+  )
+}
+
+export function Root({ form, companyId, children }: SignatureFormProps) {
+  useI18n('Company.SignatureForm')
+  const { onEvent, baseSubmitHandler } = useBase()
+
+  const { mutateAsync: signForm, isPending } = useSignCompanyForm(companyId)
+  const {
+    data: { document_url: pdfUrl },
+  } = useGetCompanyFormPdf(form.uuid)
+
+  const handleSubmit = async (data: SignatureFormInputs) => {
+    await baseSubmitHandler(data, async payload => {
+      const signFormResponse = await signForm({
+        form_id: form.uuid,
+        body: {
+          signature_text: payload.signature,
+          agree: payload.confirmSignature.length > 0,
+          /**
+           * TODO[GWS-3365]: This is a temporary value here until we resolve the correct way forward with the IP address
+           */
+          signed_by_ip_address: '127.0.0.1',
```

Comment:
This is something we still need to revisit. i think the consensus was that we just need to have them add the ip address via their proxy. Updated the ticket to generalize from employee document signing to company document signing

Reply:
This is probably not the solution but looks like there are a lot of tools out there that let you hit their API and they return your IP address. Maybe Gusto needs an endpoint like this? https://api.myip.com/